addon/-private/system/relationships/state/create.js

```diff
@@ -15,6 +15,8 @@ function createRelationshipFor(record, relationshipMeta, store) {
   let inverse = null;
   if (shouldFindInverse(relationshipMeta)) {
     inverse = record.type.inverseFor(relationshipMeta.key, store);
+  } else {
+    record.type.typeForRelationship(relationshipMeta.key, store);
```

Comment:
alternatively, a strippable check should be added at this location: https://github.com/emberjs/data/blob/master/addon/-private/system/relationships/ext.js#L242